client-spec/client-specification-v1.md

```diff
@@ -0,0 +1,322 @@
+
+If this field is missing, clients must assume its value to be `stable`.
+
+### ksp_version
+
+This field may contain either:
+- `any`
+- a string matching the regular expression:
+		[0-9]+\\.[0-9]+(\\.[0-9]+)?
+
+If the field contains the string `any`,
+clients must allow the installation of the module
+on any version of the game,
+unless the `ksp_version_min` and `ksp_version_max` are also present:
+in this case, the client must respect these constraints.
+
+Otherwise, the client must only allow the installation of the mod
+into a copy of the game with a matching version number.
+
+### ksp_version_min and ksp_version_max
+
+Like the `ksp_version` field, these fields can contain either the string `any`
+or a version number.
+
+If `ksp_version` is `any` and either of these fields is defined,
+the client must enforce these constraints.
+
+If `ksp_version` contains a specific version number,
+these two fields should not be present.
+In this case, the client must disregard their value and only enforce
+the most specific version constraint.
+
+
+## install stanzas
+
+Installation instructions are specified using the `install` field.
+This *optional* field contains an array of install directive
+(called "stanzas" in some of the documentation).
+Each stanza defines where to install a set of files.
+
+### Required: file and install_to
+
+If an `install` stanza is present,
+it must contain at least two fields:
+
+- `file`: path of a file or folder, relative to the top-level of the zip file obtained through the `download` link. The file (or folder) must be extracted to the location specified by the `install_to` field.
+- `install_to`: the location where the file or folder must be extracted. This field can only take one of the following 4 values:
+	. `GameRoot`: the root folder of the KSP installation;
+	. `GameData`: equal to `$(GameRoot)/GameData`;
+    . `Ships`: equal to `$(GameRoot)/Ships`
+    . `Tutorial`: equal to #TODO: where!?
+
+The client must not allow the installation of any file in any directory other than those listed above.
+The file or folder located by the `file` field must be extracted to the destination directory
+preserving the directory tree structure in the zip file.
+Installation of any file outside of the game's directory is strictly forbidden.
+
+### Optional: filters
+
+If a `filter` field is specified,
+it contains one or more files that must be excluded from the installation
+even if they are selected by the corresponding `file` field.
+The `filter` field only affects files in the same install stanza
+(that is, install stanzas are independent from each other).
+
+Analogously, if a `filter_regexp` is specified
+the client must exclude from the installation
+all the files that match one or more of the listed regular expressions.
+
+### File overwriting
+
+File overwriting is not permitted.
+Any attempt to overwrite another file
+must result in an error,
+aborting the installation attempt.
+
+### Rolling back
```

Comment:
Clients *must* leave GameData and the registry in a consistent state. How they choose to do that is up to them, but it can include checking for file conflicts before installing anything, or they can take the more lazy approach and use transactions.

So no, you don't have to have transactions, but clients are responsible for making sure things are consistent.